packages/runtime/container-runtime/src/test/summaryManager.spec.ts

```diff
@@ -0,0 +1,356 @@
+        getDelay() { return this.delayMs; },
+        maxDelayMs: 0,
+        delayWindowMs: 0,
+        delayFn: () => 0,
+    };
+
+    class TestConnectedState extends TypedEventEmitter<IConnectedEvents> implements IConnectedState {
+        public connected = false;
+        public clientId: string | undefined;
+
+        public connect() {
+            this.connected = true;
+            this.clientId = thisClientId;
+            this.emit("connected", this.clientId);
+        }
+
+        public disconnect() {
+            this.connected = false;
+            this.emit("disconnected");
+        }
+    }
+
+    class TestSummarizer extends TypedEventEmitter<ISummarizerEvents> implements ISummarizer {
+        private notImplemented(): never {
+            throw Error("not implemented");
+        }
+        public onBehalfOf: string | undefined;
+        public state: "notStarted" | "running" | "stopped" = "notStarted";
+        public readonly stopDeferred = new Deferred<string | undefined>();
+        public readonly runDeferred = new Deferred<void>();
+
+        constructor() { super(); }
+        public async setSummarizer(): Promise<Summarizer> {
+            this.notImplemented();
+        }
+        public stop(reason?: string): void {
+            this.stopDeferred.resolve(reason);
+        }
+        public async run(onBehalfOf: string): Promise<void> {
+            this.onBehalfOf = onBehalfOf;
+            this.state = "running";
+            await Promise.all([
+                this.stopDeferred.promise,
+                this.runDeferred.promise,
+            ]);
+            this.state = "stopped";
+        }
+        public updateOnBehalfOf(onBehalfOf: string): void {
+            this.onBehalfOf = onBehalfOf;
+        }
+
+        public readonly summarizeOnDemand: ISummarizer["summarizeOnDemand"] = () => this.notImplemented();
+        public readonly enqueueSummarize: ISummarizer["enqueueSummarize"] = () => this.notImplemented();
+        public readonly request: ISummarizer["request"] = () => this.notImplemented();
+        public get IFluidLoadable(): IFluidLoadable { return this.notImplemented(); }
+        public get IFluidRouter(): IFluidRouter { return this.notImplemented(); }
+        public get handle(): IFluidHandle { return this.notImplemented(); }
+    }
+
+    class TestSummarizerClientElection
+        extends TypedEventEmitter<ISummarizerClientElectionEvents>
+        implements ISummarizerClientElection {
+        public electedClientId: string | undefined;
+
+        public electClient(clientId: string | undefined) {
+            this.electedClientId = clientId;
+            this.emit("electedSummarizerChanged");
+        }
+    }
+
+    let clientElection: TestSummarizerClientElection;
+    let connectedState: TestConnectedState;
+    let summarizer: TestSummarizer;
+    let requestCalls = 0;
+    let requestDeferred = new Deferred<void>();
+    const requestSummarizer = async (): Promise<ISummarizer> => {
```

Comment:
Consider adding a comment for this function for things which are not very obvious:
- requestCalls is incremented every time SummaryManager requests a summarizer and we validate the the this is called / not called as expected.
- requestDeffered is used to delay returning the summarizer so that we can test "requested" but still "not running" state of the SummaryManager.